TokenCacheMigration/ADAL2MSAL/AdalFileCache.cs

```diff
@@ -1,51 +1,33 @@
-﻿/*
- The MIT License (MIT)
-
-Copyright (c) 2015 Microsoft Corporation
-
-Permission is hereby granted, free of charge, to any person obtaining a copy
-of this software and associated documentation files (the "Software"), to deal
-in the Software without restriction, including without limitation the rights
-to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
-copies of the Software, and to permit persons to whom the Software is
-furnished to do so, subject to the following conditions:
-
-The above copyright notice and this permission notice shall be included in all
-copies or substantial portions of the Software.
-
-THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
-IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
-FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
-AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
-LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
-OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
-SOFTWARE.
-*/
-
-using Microsoft.Identity.Core.Cache;
-using Microsoft.IdentityModel.Clients.ActiveDirectory;
+﻿using Microsoft.IdentityModel.Clients.ActiveDirectory;
+using System;
+using System.Collections.Generic;
 using System.IO;
+using System.Linq;
 using System.Security.Cryptography;
+using System.Text;
+using System.Threading.Tasks;
 
-namespace CommonCacheADALV5
+namespace ADAL2MSAL
 {
     /// <summary>
-    /// Simple file based persistent cache implementation for a desktop application (from ADAL 4.x)
+    /// IMPORTANT: Only suitable for desktop apps (public client apps)!
     /// </summary>
-    class FilesBasedTokenCache : TokenCache
+    /// <remarks>
+    /// In ADAL v3, some of the APIs are named slightly differently. See https://github.com/Azure-Samples/active-directory-dotnet-v1-to-v2/blob/master/TokenCacheMigration/CommonCacheAdalv3/FilesBasedAdalV3TokenCache.cs
+    /// </remarks>
+    class AdalFileCache : TokenCache
```

Comment:
I think this is the most important simplification: 

- ADAL no longer uses MSAL cache format